File: frameworks/Scala/play2-scala/play2-scala/app/controllers/Application.scala
package controllers

import javax.inject.{ Inject, Singleton }

import play.api.mvc._
import play.api.libs.json.Json
import play.mvc.Http

@Singleton
class Application @Inject() (cc: ControllerComponents)
extends AbstractController(cc) {

  implicit final val helloWorldWrites = Json.writes[HelloWorld]

  def getJsonMessage = Action {
    Ok( Json.toJson(HelloWorld()) )
  }

  val plaintext = Action {
    Ok("Hello, World!").as(play.api.http.MimeTypes.TEXT)
  }
}

case class HelloWorld(message: String = "Hello, World!")

File: frameworks/Java/play2-java/play2-java/app/controllers/Application.java
package controllers;

import play.libs.Json;
import play.mvc.Controller;
import play.mvc.Http.MimeTypes;
import play.mvc.Result;

public class Application extends Controller {

    public static class Message {
        public final String message = "Hello, World!";
    }

    public Result json() {
        return ok(Json.toJson(new Message()));
    }

    public Result plainText() {
        return ok("Hello, World!").as(MimeTypes.TEXT);
    }

}
